contrib/tools/config-docs-generator/Dockerfile

```diff
@@ -0,0 +1,29 @@
+# Use a specific nightly toolchain for reproducible builds
+FROM rustlang/rust@sha256:04690ffa09cddd358b349272173155319f384e57816614eea0840ec7f9422862
```

Comment:
I'm not sure how i feel about this. in theory it's good - but it would be add tech debt since it would need to be updated regularly. 

is stable not an option for this PR? if not, i would prefer if we could use a generic nightly build like `rustlang/rust:nightly` so we don't have to keep updating this (ideally) rarely used Dockerfile. 

another option is to use a default debian image and install rust nightly manually via rustup

Reply:
i see. that rationale makes a lot more sense, particularly it gives us a path to using a generic image in the future (possibly, if the nightly changes are merged to stable).